File: peak-index-in-a-mountain-array/peak-index-in-a-mountain-array.py
class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        n = len(arr)
        left = 0
        right = n-1

        while left <= right:
            mid = (left+right) // 2
            if mid < right and arr[mid] > arr[mid-1] and arr[mid+1] < arr[mid]:
                return mid
            elif mid == right:
                return mid
            elif mid < right and arr[mid] < arr[mid+1]:
                left = mid+1
            else:
                right = mid-1

        return -1

File: peak-index-in-a-mountain-array/peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        
        int start = 0, end = arr.length-1, idx = -1;
        
        
        while(start <= end){
            int mid = (start + end + 1)/2;
            
            if(arr[mid] > arr[mid+1]){
                idx = mid;
                end = mid - 1;
            }
            else{
                start = mid+1;
            }
        }
        
        return idx;
            
        
        
    }
}